File: java/app/src/main/java/cribbage/Cribbage.java
package cribbage;

import com.google.common.collect.Lists;
import cribbage.internal.Card;
import cribbage.internal.Deck;
import cribbage.internal.Hand;
import cribbage.internal.HandScorer;

import java.util.List;

public class Cribbage {

    public static void main(String[] args) {
        final Deck deck = new Deck();
        deck.shuffle();

        final List<Card> playerCards = Lists.newArrayList();
        playerCards.add(deck.deal());
        playerCards.add(deck.deal());
        playerCards.add(deck.deal());
        playerCards.add(deck.deal());

        final Card faceUpCard = deck.deal();

        final Hand hand = new Hand(playerCards, faceUpCard);

        final HandScorer scorer = new HandScorer();

        System.out.printf("Hand: %s, score: %d", hand, scorer.scoreHand(hand));
    }
}

File: scala/src/main/scala/cribbage/Cribbage.scala
package cribbage

import cribbage.internal.HandScorer.scoreHand
import cribbage.internal.domain.*

object Cribbage {

  def main(args: Array[String]): Unit = {
    val deck = Deck().shuffle()

    val (newDeck, playerCards) = deck.deal(5)

    val hiddenCards = playerCards.take(4)
    val faceUpCard = playerCards.last

    val hand = Hand(hiddenCards, faceUpCard)

    println(s"Hand: ${hand}, score: ${scoreHand(hand)}")
  }
}
